common/sgx/collateral.c

```diff
@@ -325,14 +325,21 @@ oe_result_t oe_validate_revocation_list(
                 "Failed to read CRL. %s",
                 oe_result_str(result));
         }
-        // Otherwise, CRL should have v3 structure which is Der encoded
+        // Otherwise, CRL should have v3 structure which is DER encoded
         else
         {
+            size_t der_data_size = sgx_endorsements->items[i].size;
+            if (der_data_size == 0)
+                OE_RAISE(OE_INVALID_PARAMETER);
+
+            // If DER CRL buffer has null terminator, need to remove it before
+            // send the DER data to crypto API for reading.
+            if (sgx_endorsements->items[i].data[der_data_size - 1] == 0)
```

Comment:
Do we ensure that there is at most a single trailing zero?

Reply:
Azure DCAP added the trailing zero [here](https://github.com/microsoft/Azure-DCAP-Client/blob/master/src/dcap_provider.cpp#L182) and Intel added it [here](https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteGeneration/qpl/sgx_default_quote_provider.cpp#L242).
For OE, it [changes the last byte](https://github.com/openenclave/openenclave/blob/master/host/sgx/sgxquoteprovider.c#L187) of the retrieved buffer to `\0` to make sure the buffer is NULL terminated.